src/collapse/dtrace.rs

```diff
@@ -55,24 +99,22 @@ impl Collapse for Folder {
                 break;
             }
         }
-        loop {
-            line.clear();
 
-            if reader.read_line(&mut line)? == 0 {
-                break;
-            }
+        // Do collapsing...
+        if self.opt.nthreads < 2 {
```

Comment:
```suggestion
        if self.opt.nthreads == 1 {
```

Reply:
Actually, thinking some more about it, I'd prefer to have that conditional only one place. Could we use something like `self.occurrences.is_single_threaded()`?

Reply:
Or, I guess `!self.occurrences.is_concurrent()`